src/features/query-bar/components/filter/FilterButton.js

```diff
@@ -118,6 +119,7 @@ class FilterButton extends React.Component<Props, State> {
                 onFilterChange(transientConditions);
             }
             this.setState({
+                hasAppliedFilters: true,
                 isMenuOpen: false,
                 transientConditions: [],
```

Comment:
Can you remind me why transientConditions=[] instead of null? The latter seems more appropriate.